src/Rule/LatteTemplatesRule.php

```diff
@@ -71,28 +81,49 @@ public function processNode(Node $collectedDataNode, Scope $scope): array
-                        $compileFilePath = $this->latteToPhpCompiler->compileFile($template->getActualClass(), $templatePath, $template->getVariables(), $template->getComponents());
-                    } catch (Throwable $e) {
-                        $errors = array_merge($errors, $this->errorBuilder->buildErrors([new Error($e->getMessage(), $scope->getFile())], $templatePath, $scope));
-                        continue;
-                    }
-
-                    $fileAnalyserResult = $this->fileAnalyserFactory->create()->analyseFile(
-                        $compileFilePath,
-                        [],
-                        $this->rulesRegistry,
-                        $this->collectorsRegistry,
-                        null
-                    );
-
-                    $errors = array_merge($errors, $this->errorBuilder->buildErrors($fileAnalyserResult->getErrors(), $templatePath, $scope));
-                }
+                $this->analyseTemplates($templates, $scope, $errors);
             }
         }
         return $errors;
     }
+
+    /**
+     * @param Template[] $templates
+     * @param RuleError[] $errors
+     */
+    private function analyseTemplates(array $templates, Scope $scope, array &$errors): void
+    {
+        foreach ($templates as $template) {
+            $templatePath = $template->getPath();
+
+            try {
+                $compileFilePath = $this->latteToPhpCompiler->compileFile($template->getActualClass(), $templatePath, $template->getVariables(), $template->getComponents());
+            } catch (Throwable $e) {
+                $errors = array_merge($errors, $this->errorBuilder->buildErrors([new Error($e->getMessage(), $scope->getFile())], $templatePath, $scope));
+                continue;
+            }
+
+            $fileAnalyserResult = $this->fileAnalyserFactory->create()->analyseFile(
+                $compileFilePath,
+                [],
+                $this->rulesRegistry,
+                new CollectorsRegistry([$this->includePathCollector]),
+                null
+            );
+
+            $errors = array_merge($errors, $this->errorBuilder->buildErrors($fileAnalyserResult->getErrors(), $templatePath, $scope));
+
+            $dir = dirname($templatePath);
+
+            $collectedDataList = $fileAnalyserResult->getCollectedData();
+            $includeTemplates = [];
+            foreach ($collectedDataList as $collectedData) {
+                /** @phpstan-var CollectedIncludePathArray $data */
+                $data = $collectedData->getData();
+                $collectedIncludedPath = CollectedIncludePath::fromArray($data, $this->typeStringResolver);
+                $includeTemplates[] = new Template($dir . '/' . $collectedIncludedPath->getPath(), $template->getActualClass(), $collectedIncludedPath->getVariables() + $template->getVariables(), $template->getComponents());
+            }
+            $this->analyseTemplates($includeTemplates, $scope, $errors);
```

Comment:
well, like always, there can be infinite loop. Then there is question when to stop it in static analysis. We can't stop analysing of included template just because it was already analysed (in different context)

Reply:
I would stop search for included files in files already searched for included files. So will will analyse A called from PHP, found call of B, analyse B called from A, found call of A, analyse A called from B, but we will not againg analyse A to find call of B from A again

Reply:
Also reset list for teach resolved node

Reply:
have to think about it, I'll check some projects if there is recursion possible in nette / latte itself